src/test/java/lambda/part3/exercise/Exercise2.java

```diff
@@ -0,0 +1,156 @@
+package lambda.part3.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"unused", "WeakerAccess", "ConstantConditions"})
+public class Exercise2 {
+
+    private static class MapHelper<T> {
+
+        private final List<T> source;
+
+        private MapHelper(List<T> source) {
+            this.source = source;
+        }
+
+        public static <T> MapHelper<T> from(List<T> source) {
+            return new MapHelper<>(source);
+        }
+
+        public List<T> getMapped() {
+            return new ArrayList<>(source);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается с использованием заданной функции.
+         * ([T], (T -> R)) -> [R]
+         * @param mapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> map(Function<T, R> mapping) {
+           List<R> result = new ArrayList<>();
+           source.forEach(T-> {
+               result.add(mapping.apply(T));
+           });
+            return new MapHelper<>(result);
+        }
+
+        /**
+         * Создает объект для маппинга, передавая ему новый список, построенный на основе исходного.
+         * Для добавления в новый список каждый элемент преобразовывается в список с использованием заданной функции.
+         * ([T], (T -> [R])) -> [R]
+         * @param flatMapping Функция преобразования элементов.
+         */
+        public <R> MapHelper<R> flatMap(Function<T, List<R>> flatMapping) {
+            List<List<R>> listOfListsOfR = new ArrayList<>();
+            source.forEach(T-> {
+                listOfListsOfR.add(flatMapping.apply(T));
+            });
+            List<R> result = new ArrayList<>();
+            listOfListsOfR.forEach(S-> {
+                S.forEach(L-> {
+                    result.add((R)L);
+                });
+            });
+            return new MapHelper<>(result);
```

Comment:
I'm afraid, this method is very hard to read.
I think you can make this part simplier and clearer. 

In addition, `flatMapping.apply(...)` returns list of elements, so you can use something like 
`yourList.addAll(flatMapping.apply(...)`.

And if we combine this with `source.forEach(...)` we have: 
`source.forEach(... -> yourList.addAll(flatMapping.apply(...)));`